Adds right-click options to Configuration side bar

superiorser9 opened the pull request:
Closes #13003 

Based off the issue above, I came up with the concept of right-clicking the side bar wrench (Configuration) so it will appear these options:
- "Restore default plugins" (enable enabledByDefault plugins & disable everything else including external plugins);
- "Disable all plugins" (except Configuration & Dev tools if enabled);
- "Enable all plugins" (except GPU plugins);
- "Revert changes" (this will basically revert the plugin status after using one of the above options to last stored).

I think this is more than just a master switch/using rs=vanilla , specially the "Restore default plugins" option might come in handy on discord #support.

![configoptions](https://user-images.githubusercontent.com/58041466/109054244-9384d600-76d5-11eb-93d6-672f49cf841f.gif)

abextm commented:
Disabling every plugin is a bad idea because it will make people think they can do that as a substitute to reproducing a bug on vanilla. Enabling everything is also a bad idea, as the things that are disabled by default generally have a good reason to be and may actively conflict with other plugins.
Having a restore default plugins button seems ok, but I don't think this is a good place for it. Maybe it could live in the info panel.

superiorser9 commented:
As suggested, removed everything but restore default plugins which I added to the info panel.

![image](https://user-images.githubusercontent.com/58041466/109083984-d1e2bb00-76fe-11eb-8c39-f7b8051adcc5.png)

And also a confirmation dialog:

![image](https://user-images.githubusercontent.com/58041466/109084049-ee7ef300-76fe-11eb-9513-becd5d7ed757.png)

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/info/InfoPanel.java:
```diff
@@ -170,12 +186,79 @@ void init()
+							}
+							catch (PluginInstantiationException ex)
+							{
+								log.warn("Unable to stop {}", plugin.getClass().getSimpleName());
```
This should log the exception as well. See [line 324 of PluginListPanel](https://github.com/runelite/runelite/blob/78eb69c8df8f823a590fe267bf9663591c3c09c0/runelite-client/src/main/java/net/runelite/client/plugins/config/PluginListPanel.java#L324):

```java
log.warn("Error when stopping plugin {}", plugin.getClass().getSimpleName(), ex);
```

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/info/InfoPanel.java:
```diff
@@ -170,12 +186,79 @@ void init()
+						{
+							try
+							{
+								pluginManager.setPluginEnabled(plugin, false);
```
This does not throw any exceptions, move it outside the `try` block.

superiorser9 commented on runelite-client/src/main/java/net/runelite/client/plugins/info/InfoPanel.java:
```diff
@@ -126,7 +142,7 @@ void init()
 		setLayout(new BorderLayout());
 		setBackground(ColorScheme.DARK_GRAY_COLOR);
-		setBorder(new EmptyBorder(10, 10, 10, 10));
+		setBorder(new EmptyBorder(10, 10, 10, 5));
```
setBorder(new EmptyBorder(10, 10, 10, 10))

![image](https://user-images.githubusercontent.com/58041466/109144842-9fb27700-7759-11eb-9f53-b529ac5455ef.png)

setBorder(new EmptyBorder(10, 10, 10, 5))

![image](https://user-images.githubusercontent.com/58041466/109144905-b2c54700-7759-11eb-8681-c1db58127eb0.png)

`emailLabel.setText("<a href=\"" + RUNELITE_LOGIN + "\">Login</a> to sync settings to the cloud.")` this text is too large and Box.createGlue() messes with spaces.
